text/0000-unkind-static-libraries.md

```diff
@@ -0,0 +1,89 @@
+* If I specify `kind=dynamic` `foo.lib` is passed to the linker invocations for `b.dll` and `c.exe`. `a.rlib` `b.dll` and `c.exe` will all use `dllimport` when referencing symbols from `foo.lib` and `b.dll` will __not__ `dllexport` any of the symbols from `foo.lib`.
+* If I specify `kind=better_static` `foo.lib` is passed to the linker invocation for `b.dll` but __not__ `c.exe`. `a.rlib` and `b.dll` do __not__ use `dllimport` for symbols they reference from `foo.lib`. `b.dll` will `dllexport` any symbols from `foo.lib` that are reachable and `c.exe` will `dllimport` any symbols from `foo.lib` that it uses.
+
+## Details of linking on Windows
+
+On Windows, in the MSVC world, the only kind of library you ever link to is a `foo.lib` library. This may either be a static library or an import library for a DLL (theoretically the library could contain both static symbols and dynamic imports but that is typically rare). MinGW can also sometimes link to a DLL directly.
+
+When linking to symbols from an import library, `dllimport` needs to be applied to the symbols, otherwise the generated code is less than ideal and it may even fail to link. When linking to symbols from a static library, `dllimport` should not be applied to the symbols, otherwise the generated code is less than ideal and it may even fail to link. Here is a table of various combinations of `dllimport` and their results on linking with MSVC.
+
+Library type | Static | Function | Result
+------------ | ------ | -------- | ------
+Dynamic | No | No | Success
+Dynamic | Plain | No | Error
+Dynamic | Dllimport | No | Success
+Dynamic | No | Plain | Success
+Dynamic | Plain | Plain | Error
+Dynamic | Dllimport | Plain | Success
+Dynamic | No | Dllimport | Success
+Dynamic | Plain | Dllimport | Error
+Dynamic | Dllimport | Dllimport | Success
+Static | No | No | Success
+Static | Plain | No | Success
+Static | Dllimport | No | Error
+Static | No | Plain | Success
+Static | Plain | Plain | Success
+Static | Dllimport | Plain | Warning
+Static | No | Dllimport | Error
+Static | Plain | Dllimport | Warning
+Static | Dllimport | Dllimport | Error
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+* It adds another `kind` that has to be supported and tested.
+
+# Alternatives
+[alternatives]: #alternatives
+
+* Don't do this and make me very sad.
+* Change the behavior of `kind=static`. Remove the bundling aspect and simply make it provide the knowledge to rustc that the symbols are static instead of dynamic. Since Cargo ensures the non-Rust static library will hang around until link time anyway, this would not really break anything for most people. Only a few people would be broken by this and it would be fairly easy to fix. Has the advantage of not adding another `kind`.
```

Comment:
I *much* prefer this alternative. We generally make no guarantees about the usability of anything other than executables, `cdylib`s and `staticlib`s, the last of which requires the user to make note of the library list Cargo outputs.